File: 202006/20200608/1010. 总持续时间可被 60 整除的歌曲.cc
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    int numPairsDivisibleBy60(const vector<int> &time) {
        vector<int> v(60, 0);
        for (int value : time)
            v[value % 60] += 1;
        int sum = 0;
        for (auto i = 1; i < 30; i++)
            sum += v[i] * v[60 - i];
        sum += v[0] * (v[0] - 1) / 2;
        sum += v[30] * (v[30] - 1) / 2;
        return sum;
    }
};

int main(int argc, char const *argv[]) {
    cout << Solution().numPairsDivisibleBy60({100, 100}) << endl;
    return 0;
}

File: 202006/20200608/1010. 总持续时间可被 60 整除的歌曲.py
class Solution:
    def numPairsDivisibleBy60(self, time: [int]) -> int:
        d = [0 for i in range(60)]
        for t in time:
            d[t % 60] += 1
        sum = 0
        sums = 0
        for index, value in enumerate(d):
            sums += value
            if index > 30:
                break
            if index != 30 and index != 0:
                sum += value * d[60 - index]
            else:
                sum += int(value * (value - 1) / 2)
        return sum
